hips/hip-0012.md

```diff
@@ -0,0 +1,186 @@
+All Beta and RC releases are done as part of this phase.
+
+This phase will terminate, and Helm 4.0.0 development will be deemed complete, upon the release of Helm 4.0.0.
+
+However, this phase must not be less than three months in order to give the broader ecosystem ample time to test and provide feedback.
+
+#### Halting Helm 4 Development
+
+During the course of development, it may turn out that there is not sufficient reason to continue development on Helm 4.
+For example, there may not be enough feature HIPs to justify releasing,
+or the HIPs may not achieve enough sponsorship or support from Helm project maintainers.
+
+In this case, the Helm maintainers may hold a "circuit breaker" vote to halt Helm 4 development until such a time when it is warranted.
+
+The process for halting Helm 4 development will be a simple majority vote by the Helm project maintainers.
+
+## Backwards compatibility
+
+Nothing in this document is contrary to our existing process for decision making.
+Note that this does not change the PR review process or the HIP process.
+It assumes that both of those processes will remain the same.
+
+## How to teach this
+
+The contents of this HIP will be discussed in the public Helm Dev meeting, which is recorded.
+If possible, the kick-off meeting will be held at KubeCon North America in 2021,
+and the resources of CNCF will be utilized for advertising this meeting and its purposes.
+
+During all stages of Helm 4 development, the Release Engineer will ensure that an update on progress is given at no fewer than two Helm public dev calls each month.
+
+## Rejected ideas
+
+Helm 3 was largely developed along an informal version of the above model.
+However, we were criticized for not documenting the process or explaining how we were going to follow the process.
+This proposal provides an alternative to an "informal process" that would draw similar ire.
+
+Multiple release managers was also considered.
+However, two things suggest multiple release managers would not be appropriate:
+
+1. The desire to have a single authority for architectural integrity and resolution
+2. The simple fact that there are not enough Helm core maintainers to justify having a committee
+
+We considered allowing a non-core maintainer be a release engineer, but rejected for the following reasons:
+
+1. A release engineer needs detailed information about Helm's design, intents, and implementation. Our only way to ensure this is via maintainer-ship.
+2. A non-core maintainer is not bound by the core maintainer process, and this would introduce tension and exceptionalism.
+
+It has been suggested that Microsoft should lead the development process, as it was MS engineers who started Helm, and MS engineers are the most frequent contributors to Helm.
+We have decided there is no reason why any other core maintainer should be ruled out in virtue of their chosen employment.
+We also believe such a restriction runs contrary to CNCF's intentions.
```

Comment:
Nothing against the MS engineers, all amazing folks 🤗 but this is a good direction.